sys/mips/mips/locore.S

```diff
@@ -163,9 +163,18 @@ VECTOR(_locore, unknown)
 	/*
 	 * Enable userspace cycle-count, cycle-count resolution, and CPU
 	 * number registers available via rdhwr.
+	 * Also enable beri specific statcounters.
 	 */
 	MFC0	t0, MIPS_COP_0_HWRENA
-	or	t0, MIPS_HWRENA_CC | MIPS_HWRENA_CCRES | MIPS_HWRENA_CPUNUM | (7 << 4)
+	or	t0, MIPS_HWRENA_CC | MIPS_HWRENA_CCRES | MIPS_HWRENA_CPUNUM | (7 << 4) \
+	| MIPS_BERI_HWRENA_RESET_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_ICACHE_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_DCACHE_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_L2CACHE_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_MEM_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_TAGCACHE_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_L2MASTER_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_TAGMASTER_STATCOUNTERS
```

Comment:
Ah, I missed the fact that it was in a `CPU_BERI` section already.  Are `MIPS_HWRENA_CC` and friends standard things that we happen to only support on BERI?  If not, we should rename them.  Of so, separating BERI and non-BERI things seems like a good idea even if there aren't code paths that use them.